stable/kibana/templates/configmap-dashboardimport.yaml

```diff
@@ -0,0 +1,55 @@
+{{- if .Values.ConfigmapDashboardImport.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "kibana.fullname" . }}-importscript
+  labels:
+    app: {{ template "kibana.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  dashboardimport.sh: |
+    #!/usr/bin/env bash
+    #
+    # kibana dashboard import script
+    #
+
+    cd /kibanadashboards
+
+    echo "starting kibana..."
+
+    /usr/local/bin/kibana-docker $@ &
+
+    echo "waiting up to {{ .Values.readinessProbe.initialDelaySeconds }} seconds for kibana to come up..."
+
+    for i in {1..{{ .Values.readinessProbe.initialDelaySeconds }}}; do
+      curl -Is localhost:5601/api/status | grep "HTTP.*200" > /dev/null && break || sleep 1
+    done
+
+    for DASHBOARD_FILE in *; do
+      echo -e "\nimporting ${DASHBOARD_FILE}..."
+
+      curl {{ if .Values.ConfigmapDashboardImport.xpackauth.enabled }}--user {{ .Values.ConfigmapDashboardImport.xpackauth.username }}:{{ .Values.ConfigmapDashboardImport.xpackauth.password }}{{ end }} -s --connect-timeout 60 --max-time 60 -XPOST localhost:5601/api/kibana/dashboards/import?force=true -H 'kbn-xsrf:true' -H 'Content-type:application/json' -d @./${DASHBOARD_FILE}
+
+      if [ "$?" != "0" ]; then
+        echo -e "\nimport of ${DASHBOARD_FILE} failed... restarting container..."
+        exit 1
+      else
+        echo -e "\nimport of ${DASHBOARD_FILE} finished :-)"
+      fi
+
+    done
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "kibana.fullname" . }}-dashboards
+  labels:
+    app: {{ template "kibana.name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+{{ (.Files.Glob "dashboards/*").AsConfig | indent 2 }}
+{{- end -}}
```

Comment:
Baking the dashboards into the chart is less extensible than other avenues. While I generally agree with @mattfarina 's comment regarding moving them out of the values files, this current strategy ends up hardcoding the dashboards into the chart with no extensibility for the user. Can we add an extensibility strategy here to supplement the baked in charts similar to how [grafana does](https://github.com/helm/charts/blob/master/stable/grafana/templates/dashboards-json-configmap.yaml) it? That plus the upcoming `--set-files` extension to helm would enable more extensibility.